apm-sniffer/apm-sdk-plugin/tomcat-thread-pool-plugin/src/main/java/org/apache/skywalking/apm/plugin/tomcat/thread/pool/TomcatThreadExecutorInterceptor.java

```diff
@@ -18,37 +18,40 @@
 
 package org.apache.skywalking.apm.plugin.tomcat.thread.pool;
 
-import java.util.HashMap;
-import java.util.Map;
-import java.util.concurrent.ThreadPoolExecutor;
-import java.util.function.Function;
-import java.util.function.Supplier;
 import org.apache.skywalking.apm.agent.core.meter.MeterFactory;
 import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.EnhancedInstance;
 import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.InstanceConstructorInterceptor;
+import org.apache.tomcat.util.threads.ThreadPoolExecutor;
 
 public class TomcatThreadExecutorInterceptor implements InstanceConstructorInterceptor {
 
+    private static final String METER_NAME = "thread_pool";
+    private static final String METRIC_POOL_NAME_TAG_NAME = "pool_name";
     private static final String THREAD_POOL_NAME = "tomcat_execute_pool";
-    private static final Map<String, Function<ThreadPoolExecutor, Supplier<Double>>> METRIC_MAP = new HashMap<String, Function<ThreadPoolExecutor, Supplier<Double>>>() {{
```

Comment:
@xu1009 I removed this, but hard code them in the method.(Verified by H2 case adjusted by @kezhenxu94 )

The reason is, `org.apache.tomcat.util.threads.ThreadPoolExecutor` is rewrited between `9.0.21` and `9.0.54`(don't check for more details). It doesn't **extends java.util.concurrent.ThreadPoolExecutor** anymore. Then we faced type casting exception.

<img width="1265" alt="image" src="https://user-images.githubusercontent.com/5441976/159859347-bef0145c-a877-4dd6-bc1e-8a3409bffc07.png">

Ref to 9.0.52, https://github.com/apache/tomcat/blob/9.0.52/java/org/apache/tomcat/util/threads/ThreadPoolExecutor.java

Reply:
If I use  `org.apache.tomcat.util.threads.ThreadPoolExecutor` in this static block, the JVM class loader will fail when loading this interceptor, which leaves me no choice but to move all logic inside the `onConstruct` method.